app/models/resource.rb

```diff
@@ -40,21 +40,42 @@ class Resource < RmApiResource
 
       request.get_params[:resourcetype] = filters[:type] || 'all'
 
+      titlename = filters[:name]
+      isxn = filters[:isxn]
+      subject = filters[:subject]
+      publisher = filters[:publisher]
+
+      if titlename
+        request.get_params[:search] = titlename
+        request.get_params[:searchfield] = 'titlename'
+      elsif isxn
+        request.get_params[:search] = isxn
+        request.get_params[:searchfield] = 'isxn'
+      elsif subject
+        request.get_params[:search] = subject
+        request.get_params[:searchfield] = 'subject'
+      elsif publisher
+        request.get_params[:search] = publisher
+        request.get_params[:searchfield] = 'publisher'
+      else
+        request.get_params[:search] = ''
+        request.get_params[:searchfield] = 'titlename'
+      end
+
       sort = request.get_params.delete(:sort)
       request.get_params[:orderby] =
         if sort == 'relevance'
           'relevance'
         elsif sort == 'name'
           'titlename'
         else
-          request.get_params[:q] ? 'relevance' : 'titlename'
+          request.get_params[:search].present? ? 'relevance' : 'titlename'
         end
 
-      request.get_params[:search] = request.get_params.delete(:q) || ''
-      request.get_params[:searchfield] = 'titlename'
       request.get_params[:count] ||= 25
       request.get_params[:offset] = request.get_params[:page] || 1
       request.get_params.delete(:page)
+      request.get_params.delete(:q)
```

Comment:
Do we need this, specially now that we do not have user sending `q` param anymore?

Reply:
What i was noticing is that q is still on the request params. It is nil. But still present. So what happens is the request object is built to send to RMAPI but q is still on it as nil, unless you delete it.

Also, removing line 78 causes many test failures.

Reply:
Well, I think this needs to be addressed in the UI, to not pass `q` probably? what does the complete request from UI look like? I am curious to see what's getting passed in the `q` if all we need is filter[xxx]?